File: ComponentImpl/src/main/java/com/xiaojinzi/component/impl/fragment/ModuleFragmentImpl.java
package com.xiaojinzi.component.impl.fragment;

import android.app.Application;
import android.support.annotation.NonNull;

import com.xiaojinzi.component.fragment.IComponentHostFragment;

import java.util.Collections;
import java.util.Set;

/**
 * 空实现,每一个模块的 Fragment 生成类需要继承的
 *
 * @author xiaojinzi
 */
abstract class ModuleFragmentImpl implements IComponentHostFragment {

    @Override
    public void onModuleCreate(Application app) {
    }

    @Override
    public void onModuleDestroy() {
    }

    @NonNull
    @Override
    public Set<String> getFragmentNameSet() {
        return Collections.EMPTY_SET;
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/impl/fragment/ModuleFragmentImpl.kt
package com.xiaojinzi.component.impl.fragment

import android.app.Application
import com.xiaojinzi.component.fragment.IComponentHostFragment
import java.util.*

/**
 * 空实现,每一个模块的 Fragment 生成类需要继承的
 *
 * @author xiaojinzi
 */
internal abstract class ModuleFragmentImpl : IComponentHostFragment {

    override fun onCreate(app: Application) {}
    override fun onDestroy() {}
    override val fragmentNameSet: Set<String> = emptySet()

}